packages/editor-ui/src/components/VariableSelector.vue

```diff
@@ -502,7 +531,41 @@ export default mixins(
 					const nodeConnection = this.workflow.getNodeConnectionIndexes(activeNode.name, parentNode[0], 'main');
 					const outputIndex = nodeConnection === undefined ? 0: nodeConnection.sourceIndex;
 
-					tempOutputData = this.getNodeOutputData(runData, parentNode[0], filterText, itemIndex, 0, 'main', outputIndex, true) as IVariableSelectorOption[];
+					tempOutputData = this.getNodeRunDataOutput(parentNode[0], runData, filterText, itemIndex, 0, 'main', outputIndex, true) as IVariableSelectorOption[];
+
+					let pinDataOptions: IVariableSelectorOption[] = [];
+					parentNode.forEach((parentNodeName) => {
+						const pinData = this.$store.getters['pinDataByNodeName'](parentNodeName);
+
+						if (pinData) {
+							pinDataOptions = pinDataOptions.concat(this.getNodePinDataOutput(parentNodeName, pinData, filterText, true) || []);
+						}
+					});
+
+					if (pinDataOptions.length > 0) {
```

Comment:
Out of scope: I find this hard to follow without spending longer, which makes it difficult to properly check the logic. In the future, we should make an effort to refactor this section and ideally this whole 250-line method. Since we are looking to release, now is probably not the time.

Reply:
I found it very hard to follow as well. The code there is very complicated and requires deep business-logic understanding. 

I've been doing a lot of trial-and-error work to get to understand what this does and how it interacts with the data.